File: Win/ui/views/tasks_view.py
"""
TasksView.py — DAS CRM Windows
Task Management with Priority, Assignment, and Due Date Tracking
Feature parity with Android TasksScreen.tsx
"""

from PyQt6.QtWidgets import (
    QWidget, QVBoxLayout, QHBoxLayout, QLabel, QPushButton, QLineEdit,
    QScrollArea, QFrame, QTableWidget, QTableWidgetItem, QAbstractItemView,
    QMessageBox, QDialog, QComboBox, QDateEdit, QCheckBox
)
from PyQt6.QtCore import Qt, QDate
from PyQt6.QtGui import QFont, QBrush, QColor
from dataclasses import dataclass

# ─────────────────────────────────────────────────────────────────────────────────────
# DATA MODELS
# ─────────────────────────────────────────────────────────────────────────────────────

@dataclass
class TaskItem:
    """Represents a task"""
    id: str
    title: str
    description: str
    assignedTo: str
    dueDate: str
    priority: str  # HIGH, MEDIUM, LOW
    status: str  # TODO, IN_PROGRESS, COMPLETED
    category: str  # FOLLOW_UP, PROPOSAL, MEETING, OTHER

PRIORITY_OPTIONS = ["HIGH", "MEDIUM", "LOW"]
STATUS_OPTIONS = ["TODO", "IN_PROGRESS", "COMPLETED"]
CATEGORY_OPTIONS = ["FOLLOW_UP", "PROPOSAL", "MEETING", "OTHER"]

FALLBACK_TASKS = [
    TaskItem("t1", "Follow up with TechCorp", "Call Rajesh regarding proposal feedback",
            "Priya Sharma", "2026-08-27", "HIGH", "TODO", "FOLLOW_UP"),
    TaskItem("t2", "Prepare Global Solutions proposal", "Complete pricing and scope document",
            "Vikram Mehta", "2026-08-29", "HIGH", "IN_PROGRESS", "PROPOSAL"),
    TaskItem("t3", "Schedule demo with FastTrack", "Confirm meeting time with client",
            "Amit Patel", "2026-08-28", "MEDIUM", "TODO", "MEETING"),
    TaskItem("t4", "Review contract terms", "Legal review of Premium Partners agreement",
            "Sunita Rao", "2026-08-30", "HIGH", "IN_PROGRESS", "OTHER"),
    TaskItem("t5", "Send invoice to Regional Corp", "Invoice for completed integration project",
            "Rajesh Kumar", "2026-08-26", "MEDIUM", "COMPLETED", "OTHER"),
    TaskItem("t6", "Update CRM with new leads", "Import leads from latest CSV upload",
            "Priya Sharma", "2026-08-27", "LOW", "TODO", "FOLLOW_UP"),
]

# ─────────────────────────────────────────────────────────────────────────────────────
# TASK DETAILS MODAL
# ─────────────────────────────────────────────────────────────────────────────────────

class TaskDetailsModal(QDialog):
    """Modal showing detailed task information"""
    def __init__(self, task: TaskItem, parent=None):
        super().__init__(parent)
        self.setWindowTitle(f"✓ Task Details - {task.title}")
        self.setGeometry(100, 100, 550, 500)
        self.setStyleSheet("""
            QDialog { background-color: #0f172a; }
            QLabel { color: #f8fafc; }
            QPushButton { padding: 8px 12px; border-radius: 6px; font-weight: bold; }
            QPushButton#edit { background-color: #f97316; color: white; }
            QPushButton#complete { background-color: #10b981; color: white; }
            QPushButton#close { background-color: #1e293b; color: #94a3b8; }
        """)

        layout = QVBoxLayout(self)

        # Header
        headerLayout = QHBoxLayout()
        headerLayout.setContentsMargins(16, 16, 16, 12)

        titleLabel = QLabel(f"✓ {task.title}")
        titleLabel.setFont(QFont("Segoe UI", 14, QFont.Weight.Bold))
        titleLabel.setStyleSheet("color: #ffffff;")
        headerLayout.addWidget(titleLabel)
        headerLayout.addStretch()

        statusLabel = QLabel(task.status)
        statusLabel.setFont(QFont("Segoe UI", 9, QFont.Weight.Bold))
        statusColor = "#34d399" if task.status == "COMPLETED" else "#38bdf8" if task.status == "IN_PROGRESS" else "#fbbf24"
        statusLabel.setStyleSheet(f"""
            background-color: rgba(100, 100, 100, 0.15);
            color: {statusColor};
            padding: 4px 8px;
            border: 1px solid {statusColor};
            border-radius: 6px;
        """)
        headerLayout.addWidget(statusLabel)

        layout.addLayout(headerLayout)

        # Content
        contentLayout = QVBoxLayout()
        contentLayout.setContentsMargins(16, 0, 16, 12)
        contentLayout.setSpacing(12)

        # Task Info
        infoCard = self._build_info_section(
            "📋 Task Information",
            [
                ("Title", task.title),
                ("Description", task.description),
                ("Category", task.category),
                ("Priority", task.priority),
            ]
        )
        contentLayout.addWidget(infoCard)

        # Assignment
        assignCard = self._build_info_section(
            "👤 Assignment",
            [
                ("Assigned To", task.assignedTo),
                ("Due Date", task.dueDate),
                ("Status", task.status),
            ]
        )
        contentLayout.addWidget(assignCard)

        contentLayout.addStretch()

        layout.addLayout(contentLayout, 1)

        # Action Buttons
        actionLayout = QHBoxLayout()
        actionLayout.setContentsMargins(16, 0, 16, 16)
        actionLayout.setSpacing(8)

        btnEdit = QPushButton("✏️ Edit")
        btnEdit.setObjectName("edit")
        btnEdit.clicked.connect(lambda: QMessageBox.information(self, "Edit", f"Editing {task.title}..."))
        actionLayout.addWidget(btnEdit)

        btnComplete = QPushButton("✅ Mark Complete")
        btnComplete.setObjectName("complete")
        btnComplete.clicked.connect(lambda: QMessageBox.information(self, "Complete", f"Marked {task.title} as complete."))
        actionLayout.addWidget(btnComplete)

        actionLayout.addStretch()

        btnClose = QPushButton("Close")
        btnClose.setObjectName("close")
        btnClose.clicked.connect(self.accept)
        actionLayout.addWidget(btnClose)

        layout.addLayout(actionLayout)

    def _build_info_section(self, title: str, fields: list) -> QFrame:
        """Build an info section card"""
        card = QFrame()
        card.setStyleSheet("""
            QFrame {
                background-color: #0f172a;
                border: 1px solid #1e293b;
                border-radius: 12px;
                padding: 12px;
            }
        """)

        layout = QVBoxLayout(card)
        layout.setContentsMargins(12, 12, 12, 12)
        layout.setSpacing(8)

        titleLabel = QLabel(title)
        titleLabel.setFont(QFont("Segoe UI", 11, QFont.Weight.Bold))
        titleLabel.setStyleSheet("color: #ffffff;")
        layout.addWidget(titleLabel)

        for fieldName, fieldValue in fields:
            fieldLayout = QHBoxLayout()

            nameLabel = QLabel(f"{fieldName}:")
            nameLabel.setFont(QFont("Segoe UI", 10, QFont.Weight.Bold))
            nameLabel.setStyleSheet("color: #94a3b8;")
            nameLabel.setMaximumWidth(100)

            valueLabel = QLabel(fieldValue)
            valueLabel.setFont(QFont("Segoe UI", 10))
            valueLabel.setStyleSheet("color: #cbd5e1;")
            valueLabel.setWordWrap(True)

            fieldLayout.addWidget(nameLabel)
            fieldLayout.addWidget(valueLabel, 1)
            layout.addLayout(fieldLayout)

        return card

# ─────────────────────────────────────────────────────────────────────────────────────
# CREATE TASK MODAL
# ─────────────────────────────────────────────────────────────────────────────────────

class CreateTaskModal(QDialog):
    """Modal for creating new task"""
    def __init__(self, parent=None):
        super().__init__(parent)
        self.setWindowTitle("➕ Create New Task")
        self.setGeometry(100, 100, 500, 550)
        self.setStyleSheet("""
            QDialog { background-color: #0f172a; }
            QLabel { color: #cbd5e1; font-weight: bold; font-size: 10px; }
            QLineEdit, QComboBox, QDateEdit { background-color: #020617; color: #ffffff;
                                             border: 1px solid #334155; border-radius: 6px; padding: 6px; }
            QPushButton#create { background-color: #10b981; color: white; }
            QPushButton#cancel { background-color: #1e293b; color: #94a3b8; }
        """)

        layout = QVBoxLayout(self)

        title = QLabel("➕ Create New Task")
        title.setFont(QFont("Segoe UI", 13, QFont.Weight.Bold))
        title.setStyleSheet("color: #ffffff;")
        layout.addWidget(title)

        desc = QLabel("Create a new task and assign it to a team member.")
        desc.setStyleSheet("color: #94a3b8; margin-bottom: 12px;")
        layout.addWidget(desc)

        # Form fields
        layout.addWidget(QLabel("Task Title *"))
        self.titleInput = QLineEdit()
        self.titleInput.setPlaceholderText("e.g. Follow up with client")
        layout.addWidget(self.titleInput)

        layout.addWidget(QLabel("Description"))
        self.descInput = QLineEdit()
        self.descInput.setPlaceholderText("Task details...")
        layout.addWidget(self.descInput)

        layout.addWidget(QLabel("Category *"))
        self.categoryCombo = QComboBox()
        self.categoryCombo.addItems(CATEGORY_OPTIONS)
        layout.addWidget(self.categoryCombo)

        layout.addWidget(QLabel("Priority *"))
        self.priorityCombo = QComboBox()
        self.priorityCombo.addItems(PRIORITY_OPTIONS)
        layout.addWidget(self.priorityCombo)

        layout.addWidget(QLabel("Assign To *"))
        self.assignedInput = QLineEdit()
        self.assignedInput.setPlaceholderText("e.g. Rajesh Kumar")
        layout.addWidget(self.assignedInput)

        layout.addWidget(QLabel("Due Date *"))
        self.dueDateInput = QDateEdit()
        self.dueDateInput.setDate(QDate.currentDate())
        layout.addWidget(self.dueDateInput)

        layout.addStretch()

        btnLayout = QHBoxLayout()
        btnCancel = QPushButton("Cancel")
        btnCancel.setObjectName("cancel")
        btnCancel.clicked.connect(self.reject)
        btnCreate = QPushButton("Create Task ✓")
        btnCreate.setObjectName("create")
        btnCreate.clicked.connect(self.accept)
        btnLayout.addWidget(btnCancel, 1)
        btnLayout.addWidget(btnCreate, 1)
        layout.addLayout(btnLayout)

    def get_task(self) -> TaskItem:
        """Return created task"""
        return TaskItem(
            id=f"t-{id(self)}",
            title=self.titleInput.text().strip(),
            description=self.descInput.text().strip(),
            assignedTo=self.assignedInput.text().strip(),
            dueDate=self.dueDateInput.date().toString("yyyy-MM-dd"),
            priority=self.priorityCombo.currentText(),
            status="TODO",
            category=self.categoryCombo.currentText()
        )

# ─────────────────────────────────────────────────────────────────────────────────────
# MAIN TASKS VIEW
# ─────────────────────────────────────────────────────────────────────────────────────

class TasksView(QWidget):
    """Task Management with Priority & Due Date Tracking"""

    def __init__(self, parent=None):
        super().__init__(parent)
        self.setStyleSheet("""
            QWidget { background-color: #090d16; }
            QLabel { color: #f8fafc; }
            QPushButton { color: white; font-weight: bold; border: none; border-radius: 6px; }
            QLineEdit { background-color: #0f172a; color: #ffffff; border: 1px solid #1e293b;
                       border-radius: 6px; padding: 8px; }
        """)

        self.tasksList = list(FALLBACK_TASKS)
        self.search = ""
        self.selectedStatus = "ALL"

        self._build_ui()

    def _build_ui(self):
        """Build tasks UI"""
        layout = QVBoxLayout(self)
        layout.setContentsMargins(0, 0, 0, 0)

        scrollArea = QScrollArea()
        scrollArea.setWidgetResizable(True)
        scrollArea.setStyleSheet("QScrollArea { border: none; background-color: #090d16; }")

        scrollWidget = QWidget()
        scrollLayout = QVBoxLayout(scrollWidget)
        scrollLayout.setContentsMargins(16, 16, 16, 24)
        scrollLayout.setSpacing(12)

        # Title
        titleLabel = QLabel("✓ Tasks & Follow-ups")
        titleLabel.setFont(QFont("Segoe UI", 14, QFont.Weight.Bold))
        titleLabel.setStyleSheet("color: #ffffff;")
        scrollLayout.addWidget(titleLabel)

        # Search & Filter Bar
        searchLayout = QVBoxLayout()
        searchLayout.setContentsMargins(0, 0, 0, 0)
        searchLayout.setSpacing(8)

        # Search input
        self.searchInput = QLineEdit()
        self.searchInput.setPlaceholderText("🔍 Search by task title, assignee...")
        self.searchInput.setMinimumHeight(32)
        self.searchInput.textChanged.connect(self._on_search_changed)
        searchLayout.addWidget(self.searchInput)

        # Action buttons
        actionLayout = QHBoxLayout()
        btnAdd = QPushButton("➕ New Task")
        btnAdd.setStyleSheet("background-color: #10b981; padding: 6px 12px;")
        btnAdd.clicked.connect(self._open_create_task)
        actionLayout.addWidget(btnAdd)

        actionLayout.addStretch()
        searchLayout.addLayout(actionLayout)

        # Status filter chips
        statusLayout = QHBoxLayout()
        statuses = ["ALL"] + STATUS_OPTIONS
        for status in statuses:
            btn = QPushButton(status)
            btn.setCheckable(True)
            btn.setChecked(status == "ALL")
            btn.setMaximumHeight(24)
            btn.setStyleSheet(f"""
                QPushButton {{
                    background-color: #020617;
                    border: 1px solid #1e293b;
                    color: #94a3b8;
                    padding: 4px 8px;
                    border-radius: 6px;
                    font-size: 9px;
                    font-weight: 800;
                }}
                QPushButton:checked {{
                    background-color: #4f46e5;
                    color: #ffffff;
                    border-color: #4f46e5;
                }}
            """)
            btn.toggled.connect(lambda checked, s=status: self._set_status_filter(s) if checked else None)
            statusLayout.addWidget(btn)

        statusLayout.addStretch()
        searchLayout.addLayout(statusLayout)

        scrollLayout.addLayout(searchLayout)

        # Tasks Table
        self.tasksTable = QTableWidget()
        self.tasksTable.setColumnCount(7)
        self.tasksTable.setHorizontalHeaderLabels([
            "Task", "Category", "Priority", "Assigned To", "Due Date", "Status", "Action"
        ])
        self.tasksTable.horizontalHeader().setStretchLastSection(False)
        self.tasksTable.setSelectionBehavior(QAbstractItemView.SelectionBehavior.SelectRows)
        self.tasksTable.setSelectionMode(QAbstractItemView.SelectionMode.SingleSelection)
        self.tasksTable.setStyleSheet("""
            QTableWidget { background-color: #030712; gridline-color: #1e293b; }
            QHeaderView::section { background-color: #0b1329; color: #818cf8; padding: 6px;
                                   border: none; border-right: 1px solid #1e293b; }
        """)

        self.tasksTable.setColumnWidth(0, 150)
        self.tasksTable.setColumnWidth(1, 110)
        self.tasksTable.setColumnWidth(2, 90)
        self.tasksTable.setColumnWidth(3, 120)
        self.tasksTable.setColumnWidth(4, 100)
        self.tasksTable.setColumnWidth(5, 100)
        self.tasksTable.setColumnWidth(6, 80)

        self.tasksTable.doubleClicked.connect(self._open_task_details)

        self._refresh_tasks_table()

        scrollLayout.addWidget(self.tasksTable, 1)

        scrollArea.setWidget(scrollWidget)
        layout.addWidget(scrollArea)

    def _refresh_tasks_table(self):
        """Refresh tasks table"""
        filtered = self._get_filtered_tasks()

        self.tasksTable.setRowCount(len(filtered))

        for rowIdx, task in enumerate(filtered):
            self.tasksTable.setItem(rowIdx, 0, QTableWidgetItem(task.title))
            self.tasksTable.setItem(rowIdx, 1, QTableWidgetItem(task.category))

            priorityItem = QTableWidgetItem(task.priority)
            priorityColor = "#ef4444" if task.priority == "HIGH" else "#fbbf24" if task.priority == "MEDIUM" else "#34d399"
            priorityItem.setForeground(QBrush(QColor(priorityColor)))
            self.tasksTable.setItem(rowIdx, 2, priorityItem)

            self.tasksTable.setItem(rowIdx, 3, QTableWidgetItem(task.assignedTo))
            self.tasksTable.setItem(rowIdx, 4, QTableWidgetItem(task.dueDate))

            statusItem = QTableWidgetItem(task.status)
            statusColor = "#34d399" if task.status == "COMPLETED" else "#38bdf8" if task.status == "IN_PROGRESS" else "#fbbf24"
            statusItem.setForeground(QBrush(QColor(statusColor)))
            self.tasksTable.setItem(rowIdx, 5, statusItem)

            viewBtn = QPushButton("👁️ View")
            viewBtn.setStyleSheet("background-color: #4f46e5; padding: 4px 8px; font-size: 9px;")
            viewBtn.clicked.connect(lambda checked, t=task: self._open_task_details_for(t))
            self.tasksTable.setCellWidget(rowIdx, 6, viewBtn)

    def _get_filtered_tasks(self) -> list:
        """Get filtered tasks"""
        result = []

        for task in self.tasksList:
            # Status filter
            if self.selectedStatus != "ALL" and task.status != self.selectedStatus:
                continue

            # Search filter
            if self.search.strip():
                q = self.search.lower()
                matches = (
                    q in task.title.lower() or
                    q in task.assignedTo.lower() or
                    q in task.category.lower()
                )
                if not matches:
                    continue

            result.append(task)

        return result

    def _on_search_changed(self):
        """Handle search input changed"""
        self.search = self.searchInput.text()
        self._refresh_tasks_table()

    def _set_status_filter(self, status: str):
        """Set status filter"""
        self.selectedStatus = status
        self._refresh_tasks_table()

    def _open_create_task(self):
        """Open create task modal"""
        dialog = CreateTaskModal(self)
        if dialog.exec() == QDialog.DialogCode.Accepted:
            task = dialog.get_task()
            self.tasksList.insert(0, task)
            self._refresh_tasks_table()
            QMessageBox.information(self, "✓ Task Created", f"Created task: {task.title}")

    def _open_task_details(self, index):
        """Open task details modal"""
        row = index.row()
        filtered = self._get_filtered_tasks()

        if row < len(filtered):
            task = filtered[row]
            dialog = TaskDetailsModal(task, self)
            dialog.exec()

    def _open_task_details_for(self, task: TaskItem):
        """Open task details modal for specific task"""
        dialog = TaskDetailsModal(task, self)
        dialog.exec()

File: macOs/Sources/DASCRM/Views/Tasks/TasksView.swift
"""
TasksView.swift — DAS CRM macOS
Task Management with Priority, Assignment, and Due Date Tracking
Feature parity with Android TasksScreen.tsx
"""

import SwiftUI

struct TaskItem: Identifiable {
    let id: String
    let title: String
    let description: String
    let assignedTo: String
    let dueDate: String
    let priority: String
    let status: String
    let category: String
}

let taskStatuses = ["TODO", "IN_PROGRESS", "COMPLETED"]
let taskCategories = ["FOLLOW_UP", "PROPOSAL", "MEETING", "OTHER"]
let taskPriorities = ["HIGH", "MEDIUM", "LOW"]

let fallbackTasks = [
    TaskItem(id: "t1", title: "Follow up with TechCorp", description: "Call Rajesh regarding proposal feedback",
            assignedTo: "Priya Sharma", dueDate: "2026-08-27", priority: "HIGH", status: "TODO", category: "FOLLOW_UP"),
    TaskItem(id: "t2", title: "Prepare Global Solutions proposal", description: "Complete pricing and scope document",
            assignedTo: "Vikram Mehta", dueDate: "2026-08-29", priority: "HIGH", status: "IN_PROGRESS", category: "PROPOSAL"),
    TaskItem(id: "t3", title: "Schedule demo with FastTrack", description: "Confirm meeting time with client",
            assignedTo: "Amit Patel", dueDate: "2026-08-28", priority: "MEDIUM", status: "TODO", category: "MEETING"),
    TaskItem(id: "t4", title: "Review contract terms", description: "Legal review of Premium Partners agreement",
            assignedTo: "Sunita Rao", dueDate: "2026-08-30", priority: "HIGH", status: "IN_PROGRESS", category: "OTHER"),
    TaskItem(id: "t5", title: "Send invoice to Regional Corp", description: "Invoice for completed integration project",
            assignedTo: "Rajesh Kumar", dueDate: "2026-08-26", priority: "MEDIUM", status: "COMPLETED", category: "OTHER"),
    TaskItem(id: "t6", title: "Update CRM with new leads", description: "Import leads from latest CSV upload",
            assignedTo: "Priya Sharma", dueDate: "2026-08-27", priority: "LOW", status: "TODO", category: "FOLLOW_UP"),
]

@MainActor
class TasksViewModel: ObservableObject {
    @Published var tasks: [TaskItem] = fallbackTasks
    @Published var search: String = ""
    @Published var selectedStatus: String = "ALL"

    var filteredTasks: [TaskItem] {
        tasks.filter { task in
            let passesStatusFilter = selectedStatus == "ALL" || task.status == selectedStatus
            let passesSearch: Bool
            if search.isEmpty {
                passesSearch = true
            } else {
                let q = search.lowercased()
                passesSearch = task.title.lowercased().contains(q) ||
                              task.assignedTo.lowercased().contains(q) ||
                              task.category.lowercased().contains(q)
            }
            return passesStatusFilter && passesSearch
        }
    }
}

struct TasksView: View {
    @StateObject private var viewModel = TasksViewModel()
    @State private var selectedTask: TaskItem?
    @State private var showTaskDetails = false
    @State private var showCreateTask = false

    var body: some View {
        VStack(alignment: .leading, spacing: 0) {
            // Header
            VStack(alignment: .leading, spacing: 12) {
                Text("✓ Tasks & Follow-ups")
                    .font(.system(size: 14, weight: .bold))
                    .foregroundColor(.white)

                // Search input
                HStack(spacing: 8) {
                    Image(systemName: "magnifyingglass")
                        .foregroundColor(Color(red: 0.40, green: 0.45, blue: 0.52))
                    TextField("🔍 Search by task title, assignee...", text: $viewModel.search)
                        .textFieldStyle(.plain)
                        .foregroundColor(.white)

                    if !viewModel.search.isEmpty {
                        Button(action: { viewModel.search = "" }) {
                            Image(systemName: "xmark.circle.fill")
                                .foregroundColor(Color(red: 0.58, green: 0.64, blue: 0.68))
                        }
                        .buttonStyle(.plain)
                    }
                }
                .padding(.horizontal, 12)
                .padding(.vertical, 8)
                .background(Color(red: 0.06, green: 0.09, blue: 0.16))
                .border(Color(red: 0.12, green: 0.16, blue: 0.23), width: 1)
                .cornerRadius(8)

                // Action buttons
                HStack {
                    Button(action: { showCreateTask = true }) {
                        Text("➕ New Task")
                            .font(.system(size: 10, weight: .bold))
                            .foregroundColor(.white)
                            .padding(.horizontal, 12)
                            .padding(.vertical, 6)
                            .background(Color(red: 0.06, green: 0.68, blue: 0.50))
                            .cornerRadius(6)
                    }
                    .buttonStyle(.plain)

                    Spacer()
                }

                // Status filter chips
                HStack(spacing: 6) {
                    ForEach(["ALL"] + taskStatuses, id: \.self) { status in
                        Button(action: { viewModel.selectedStatus = status }) {
                            Text(status)
                                .font(.system(size: 9, weight: .bold))
                                .foregroundColor(viewModel.selectedStatus == status ? .white : Color(red: 0.58, green: 0.64, blue: 0.68))
                        }
                        .buttonStyle(.plain)
                        .padding(.horizontal, 8)
                        .padding(.vertical, 4)
                        .background(viewModel.selectedStatus == status ? Color(red: 0.31, green: 0.27, blue: 0.90) : Color(red: 0.02, green: 0.06, blue: 0.12))
                        .border(viewModel.selectedStatus == status ? Color(red: 0.31, green: 0.27, blue: 0.90) : Color(red: 0.12, green: 0.16, blue: 0.23), width: 1)
                        .cornerRadius(6)
                    }
                    Spacer()
                }
            }
            .padding(16)
            .background(Color(red: 0.06, green: 0.09, blue: 0.16))
            .border(Color(red: 0.12, green: 0.16, blue: 0.23), width: 1)

            // Tasks Table
            Table(viewModel.filteredTasks) {
                TableColumn("Task", value: \.title)
                TableColumn("Category", value: \.category)
                TableColumn("Priority") { task in
                    Text(task.priority)
                        .foregroundColor(
                            task.priority == "HIGH" ? Color(red: 0.98, green: 0.30, blue: 0.40) :
                            task.priority == "MEDIUM" ? Color(red: 0.98, green: 0.75, blue: 0.14) :
                            Color(red: 0.2, green: 0.83, blue: 0.60)
                        )
                }
                TableColumn("Assigned To", value: \.assignedTo)
                TableColumn("Due Date", value: \.dueDate)
                TableColumn("Status") { task in
                    Text(task.status)
                        .foregroundColor(
                            task.status == "COMPLETED" ? Color(red: 0.2, green: 0.83, blue: 0.60) :
                            task.status == "IN_PROGRESS" ? Color(red: 0.38, green: 0.65, blue: 0.98) :
                            Color(red: 0.98, green: 0.75, blue: 0.14)
                        )
                }
                TableColumn("Action") { task in
                    Button(action: {
                        selectedTask = task
                        showTaskDetails = true
                    }) {
                        Text("👁️ View")
                            .font(.system(size: 9, weight: .bold))
                            .foregroundColor(.white)
                            .padding(.horizontal, 8)
                            .padding(.vertical, 4)
                            .background(Color(red: 0.31, green: 0.27, blue: 0.90))
                            .cornerRadius(4)
                    }
                    .buttonStyle(.plain)
                }
            }
            .background(Color(red: 0.03, green: 0.04, blue: 0.07))
            .onDoubleClickSelectAll(false)
        }
        .background(Color(red: 0.06, green: 0.08, blue: 0.14))
        .navigationTitle("Tasks")
        .sheet(isPresented: $showTaskDetails) {
            if let task = selectedTask {
                TaskDetailsSheet(task: task, isPresented: $showTaskDetails)
            }
        }
        .sheet(isPresented: $showCreateTask) {
            CreateTaskSheet(tasks: $viewModel.tasks, isPresented: $showCreateTask)
        }
    }
}

struct TaskDetailsSheet: View {
    let task: TaskItem
    @Binding var isPresented: Bool
    @State private var showEditAlert = false
    @State private var showCompleteAlert = false

    var body: some View {
        NavigationView {
            ScrollView {
                VStack(alignment: .leading, spacing: 12) {
                    // Header
                    HStack {
                        VStack(alignment: .leading, spacing: 4) {
                            Text("✓ \(task.title)")
                                .font(.system(size: 16, weight: .bold))
                                .foregroundColor(.white)
                        }
                        Spacer()

                        Text(task.status)
                            .font(.system(size: 9, weight: .bold))
                            .foregroundColor(
                                task.status == "COMPLETED" ? Color(red: 0.2, green: 0.83, blue: 0.60) :
                                task.status == "IN_PROGRESS" ? Color(red: 0.38, green: 0.65, blue: 0.98) :
                                Color(red: 0.98, green: 0.75, blue: 0.14)
                            )
                            .padding(.horizontal, 8)
                            .padding(.vertical, 4)
                            .background(
                                (task.status == "COMPLETED" ? Color(red: 0.2, green: 0.83, blue: 0.60) :
                                task.status == "IN_PROGRESS" ? Color(red: 0.38, green: 0.65, blue: 0.98) :
                                Color(red: 0.98, green: 0.75, blue: 0.14)).opacity(0.15)
                            )
                            .cornerRadius(4)
                    }
                    .padding(12)
                    .background(Color(red: 0.06, green: 0.09, blue: 0.16))
                    .border(Color(red: 0.12, green: 0.16, blue: 0.23), width: 1)
                    .cornerRadius(12)

                    // Task Info
                    VStack(alignment: .leading, spacing: 8) {
                        Text("📋 Task Information")
                            .font(.system(size: 11, weight: .bold))
                            .foregroundColor(.white)

                        DetailRow(label: "Title", value: task.title)
                        DetailRow(label: "Description", value: task.description)
                        DetailRow(label: "Category", value: task.category)
                        DetailRow(label: "Priority", value: task.priority)
                    }
                    .padding(12)
                    .background(Color(red: 0.06, green: 0.09, blue: 0.16))
                    .border(Color(red: 0.12, green: 0.16, blue: 0.23), width: 1)
                    .cornerRadius(12)

                    // Assignment
                    VStack(alignment: .leading, spacing: 8) {
                        Text("👤 Assignment")
                            .font(.system(size: 11, weight: .bold))
                            .foregroundColor(.white)

                        DetailRow(label: "Assigned To", value: task.assignedTo)
                        DetailRow(label: "Due Date", value: task.dueDate)
                    }
                    .padding(12)
                    .background(Color(red: 0.06, green: 0.09, blue: 0.16))
                    .border(Color(red: 0.12, green: 0.16, blue: 0.23), width: 1)
                    .cornerRadius(12)

                    Spacer()

                    // Action Buttons
                    HStack(spacing: 8) {
                        Button(action: { showEditAlert = true }) {
                            Text("✏️ Edit")
                                .font(.system(size: 10, weight: .bold))
                                .foregroundColor(.white)
                                .frame(maxWidth: .infinity)
                                .padding(8)
                                .background(Color(red: 0.98, green: 0.58, blue: 0.09))
                                .cornerRadius(6)
                        }
                        Button(action: { showCompleteAlert = true }) {
                            Text("✅ Complete")
                                .font(.system(size: 10, weight: .bold))
                                .foregroundColor(.white)
                                .frame(maxWidth: .infinity)
                                .padding(8)
                                .background(Color(red: 0.06, green: 0.68, blue: 0.50))
                                .cornerRadius(6)
                        }
                    }
                }
                .padding(16)
            }
            .background(Color(red: 0.06, green: 0.08, blue: 0.14))
            .navigationTitle("Task Details")
            .navigationBarTitleDisplayMode(.inline)
            .toolbar {
                ToolbarItem(placement: .navigationBarTrailing) {
                    Button("Close") {
                        isPresented = false
                    }
                }
            }
            .alert("✏️ Edit Task", isPresented: $showEditAlert) {
                Button("OK") { }
            } message: {
                Text("Editing \(task.title)...")
            }
            .alert("✅ Complete Task", isPresented: $showCompleteAlert) {
                Button("OK") { }
            } message: {
                Text("Marking \(task.title) as complete...")
            }
        }
    }
}

struct CreateTaskSheet: View {
    @Binding var tasks: [TaskItem]
    @Binding var isPresented: Bool

    @State private var title = ""
    @State private var description = ""
    @State private var category = "FOLLOW_UP"
    @State private var priority = "MEDIUM"
    @State private var assignedTo = ""
    @State private var dueDate = Date()

    var body: some View {
        NavigationView {
            ScrollView {
                VStack(alignment: .leading, spacing: 12) {
                    Text("➕ Create New Task")
                        .font(.system(size: 13, weight: .bold))
                        .foregroundColor(.white)

                    Text("Create a new task and assign it to a team member.")
                        .font(.system(size: 10))
                        .foregroundColor(Color(red: 0.58, green: 0.64, blue: 0.68))

                    VStack(alignment: .leading, spacing: 8) {
                        Text("Task Title *")
                            .font(.system(size: 10, weight: .bold))
                            .foregroundColor(Color(red: 0.79, green: 0.84, blue: 0.88))
                        TextField("e.g. Follow up with client", text: $title)
                            .textFieldStyle(.roundedBorder)

                        Text("Description")
                            .font(.system(size: 10, weight: .bold))
                            .foregroundColor(Color(red: 0.79, green: 0.84, blue: 0.88))
                        TextField("Task details...", text: $description)
                            .textFieldStyle(.roundedBorder)

                        Text("Category *")
                            .font(.system(size: 10, weight: .bold))
                            .foregroundColor(Color(red: 0.79, green: 0.84, blue: 0.88))
                        Picker("Category", selection: $category) {
                            ForEach(taskCategories, id: \.self) { Text($0) }
                        }
                        .pickerStyle(.menu)

                        Text("Priority *")
                            .font(.system(size: 10, weight: .bold))
                            .foregroundColor(Color(red: 0.79, green: 0.84, blue: 0.88))
                        Picker("Priority", selection: $priority) {
                            ForEach(taskPriorities, id: \.self) { Text($0) }
                        }
                        .pickerStyle(.menu)

                        Text("Assign To *")
                            .font(.system(size: 10, weight: .bold))
                            .foregroundColor(Color(red: 0.79, green: 0.84, blue: 0.88))
                        TextField("e.g. Rajesh Kumar", text: $assignedTo)
                            .textFieldStyle(.roundedBorder)

                        Text("Due Date *")
                            .font(.system(size: 10, weight: .bold))
                            .foregroundColor(Color(red: 0.79, green: 0.84, blue: 0.88))
                        DatePicker("", selection: $dueDate, displayedComponents: .date)
                            .datePickerStyle(.compact)
                    }

                    Spacer()

                    HStack(spacing: 8) {
                        Button(action: { isPresented = false }) {
                            Text("Cancel")
                                .frame(maxWidth: .infinity)
                                .padding(8)
                                .background(Color(red: 0.12, green: 0.16, blue: 0.23))
                                .foregroundColor(Color(red: 0.58, green: 0.64, blue: 0.68))
                                .cornerRadius(6)
                        }
                        Button(action: {
                            let newTask = TaskItem(
                                id: "t-\(UUID())",
                                title: title,
                                description: description,
                                assignedTo: assignedTo,
                                dueDate: dueDate.formatted(date: .numeric, time: .omitted),
                                priority: priority,
                                status: "TODO",
                                category: category
                            )
                            tasks.insert(newTask, at: 0)
                            isPresented = false
                        }) {
                            Text("Create Task ✓")
                                .frame(maxWidth: .infinity)
                                .padding(8)
                                .background(Color(red: 0.06, green: 0.68, blue: 0.50))
                                .foregroundColor(.white)
                                .cornerRadius(6)
                        }
                    }
                }
                .padding(16)
            }
            .background(Color(red: 0.06, green: 0.08, blue: 0.14))
            .navigationTitle("New Task")
            .navigationBarTitleDisplayMode(.inline)
        }
    }
}

struct DetailRow: View {
    let label: String
    let value: String

    var body: some View {
        HStack {
            Text("\(label):")
                .font(.system(size: 10, weight: .bold))
                .foregroundColor(Color(red: 0.58, green: 0.64, blue: 0.68))
                .frame(width: 100, alignment: .leading)
            Text(value)
                .font(.system(size: 10))
                .foregroundColor(Color(red: 0.79, green: 0.84, blue: 0.88))
                .lineLimit(2)
            Spacer()
        }
    }
}

#Preview {
    NavigationView {
        TasksView()
    }
}
